core/src/main/java/jenkins/install/UpgradeWizard.java

```diff
@@ -0,0 +1,140 @@
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.*;
+import static org.apache.commons.io.FileUtils.*;
+import static org.apache.commons.lang.StringUtils.*;
+
+/**
+ * This is a stop-gap measure until JENKINS-33663 comes in.
+ * This call may go away. Please don't use it from outside.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+@Restricted(NoExternalUse.class)
+@Extension
+public class UpgradeWizard extends PageDecorator {
+    @Inject
+    private Jenkins jenkins;
+
+    /**
+     * Is this instance fully upgraded?
+     */
+    private volatile boolean upToDate;
+
+    /**
+     * File that captures the state of upgrade.
+     *
+     * This file records the vesrion number that the installation has upgraded to.
+     */
+    /*package*/ File getStateFile() {
+        return new File(Jenkins.getInstance().getRootDir(),"upgraded");
+    }
+
+    public UpgradeWizard() throws IOException {
+        updateUpToDate();
+    }
+
+    private void updateUpToDate() throws IOException {
+        upToDate = new VersionNumber("2.0").compareTo(getCurrentLevel()) <= 0;
+    }
+
+    /**
+     * What is the version the upgrade wizard has run the last time and upgraded to?
+     */
+    private VersionNumber getCurrentLevel() throws IOException {
+        VersionNumber from = new VersionNumber("1.0");
+        File state = getStateFile();
+        if (state.exists()) {
+            from = new VersionNumber(defaultIfBlank(readFileToString(state), "1.0").trim());
+        }
+        return from;
+    }
+
+    /**
+     * Do we need to show the upgrade wizard prompt?
+     */
+    public boolean isDue() {
+        if (upToDate)
+            return false;
+
+        // only admin users should see this
+        if (!jenkins.hasPermission(Jenkins.ADMINISTER))
+            return false;
+
+        return System.currentTimeMillis() > getStateFile().lastModified();
+    }
+
+    /**
+     * Snooze the upgrade wizard notice.
+     */
+    @RequirePOST
+    public HttpResponse doSnooze() throws IOException {
+        jenkins.checkPermission(Jenkins.ADMINISTER);
+        File f = getStateFile();
+        FileUtils.touch(f);
+        f.setLastModified(System.currentTimeMillis() + TimeUnit.DAYS.toMillis(1));
+        LOGGER.log(FINE, "Snoozed the upgrade wizard notice");
+        return HttpResponses.redirectToContextRoot();
+    }
+
+    /**
+     * Performs the upgrade activity.
+     */
+    @RequirePOST
+    public HttpResponse doUpgrade() throws IOException {
+        try {
+            if (new VersionNumber("2.0").compareTo(getCurrentLevel())>0) {
+                // 1.0 -> 2.0 upgrade
+                LOGGER.log(WARNING, "Performing 1.0 to 2.0 upgrade");
```

Comment:
It has been upgraded and it already _is_ Jenkins 2.x by this point. It's what it says in the footer. Besides plugins that aren't mandatory, nothing's different from an "upgraded" Jenkins 2.x.